corehq/apps/cloudcare/views.py

```diff
@@ -142,7 +145,12 @@ def set_cookie(response):  # set_coookie is a noop by default
             'restoreAs:{}:{}'.format(domain, request.couch_user.username))
         username = request.COOKIES.get(cookie_name)
         if username:
-            user = CouchUser.get_by_username(format_username(username, domain))
+            username = urllib.parse.unquote(username)
+            # mobile workers do not have special characters in their name
+            # and web user usernames are email addresses
+            if '@' not in username:
```

Comment:
@esoergel Can you think of a better way to distinguish between mobile and web users here?

Reply:
You could try `get_by_username` and if no user is found, then try formatting as a mobile username. I think that's a little cleaner/safer, since it doesn't depend on string formatting, but it also means a regular additional db hit.

Reply:
That would work. Since this is only called for logging in as the extra call should not be to bad.